docs/book/src/cookbook/cookbook.md

```diff
@@ -0,0 +1,32 @@
+# Cookbook
+
+A curated collection of operator development patterns, use-cases and generic\
+recommendations.
+
+## Rationale
+
+While the tutorial part of this book provides onboarding into the operator
+realm, real world use cases often surpass the tutorial example(s). Hence we
+want to provide a place to share common implementation patterns in order
+to enable other users, may they be operator experts or novices, a place to both
+share and validate common approaches they encounter while developing operators.
+
+## Recipies
+
+<aside class="note">
+
+<h1>Following Along</h1>
+
+Note that code examples of cookbook entries live as literate Go files in the
+book source directory:
+[docs/book/src/cookbook/testdata][cookbook-sources]. You may find both snippets
+and fully runnable projects among the examples.
+
+[cookbook-sources]: https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/cookbook/testdata
```

Comment:
It is an invalid link.

Reply:
To be clear, our top level `testdata` directory holds several generated projects.
We also have several `testdata` directory in `docs`. 
`testdata` is a magic name in golang, go build ignores it.
IMO it's fine to have `testdata` directory `docs` directory.

Reply:
HI @mengqiy,

So, if I understood, the idea proposed here with this PR is to create a directory where many examples and projects will start to be added by the contributors. Am I right?  And then, it means that any content added on it would be "blessed" by the project as a good practice and indeed started to be supported and maintained by it. 

Then, my concern is:  
- How much effort will be required to understand and maintain manually these examples and projects with an example provided by the community? Will it be something achievable in a medium long term? ( IHMO it is not )
- Is it the best way to improve the docs content to provide further explanations?  
- Should not better we improve the doc providing more specific examples about the subjects with explanations over it then add many projects that we and users has no context about in a dir in the repo? 
- Is it something that we really would like to persuade? 

WDYT @DirectXMan12 and @joelanford

Reply:
I am also concerned with keeping every fully functional example which could be added up to date and working. I reckon code snippets, to emphasize certain practices, embedded in the docs would be easier to maintain as, for _example_, an IDE will find references in arbitrary strings when something is refactored (rename, change of signature, move etc.)

Reply:
Before I finishing fully parsing this comment: `testdata` is just a special directory that is ignored by `go` tooling when calculating dependencies, etc.  That's all.  We put full probjects/examples in `testdata` (as well as book source code) so that it doesn't affect the dep list of the main project, and so that we can easily get the main project's tooling to ignore it most of the time.

Reply:
FWIW, I do like this idea (and we've discussed similar before).  I think we just want to keep it clear that this is a bit more tightly controlled than, say, a wiki -- we want focused, well-maintained examples.

That being said, I think literate code is the way to go, since we can check that it builds, have ide tooling as @hendrikmaus suggested, etc.

Reply:
Hi @DirectXMan12, 

what is your idea for we ensure and keep well-maintained these examples in a long-term?